locator/topology.cc

```diff
@@ -477,6 +477,17 @@ bool topology::has_node(host_id id) const noexcept {
     return bool(node);
 }
 
+const endpoint_dc_rack& topology::get_location_slow(host_id id) const {
+    // We should do the following check after lookup in nodes.
+    // In tests, there may be no config for local node, so fall back to get_location()
+    // only if no mapping is found. Otherwise, get_location() will return empty location
+    // from config or random node, neither of which is correct.
+    if (id == _cfg.this_host_id) {
+        return get_location();
+    }
+    throw std::runtime_error(format("Requested location for node {} not in topology. backtrace {}", id, lazy_backtrace()));
```

Comment:
This method is named in a confusing way, because the logic differs from `get_location(host_id)`. It doesn't get location of arbitrary node, it get's location of the local node, assuming it's equal to "id". get_location() without an argument is clear that it works on local node, but get_location_slow(host_id) still looks like it's supposed to get location of a given host_id.

Reply:
Since there is only one user, it could be inlined.